### PR TITLE
convert to typescript

### PR DESCRIPTION
**Pull request by OmgImAlexis**
Converts the whole lib to use typescript it's a little more of a change than https://github.com/lunchboxer/graphql-subscriptions-client/pull/6 but it should make it easier to keep the types up to date.

To build the js files just run `npm run build`.

### CODE REVIEW

**Comment by tom-fletcher** [comment]
Nice, I had thought about doing this but didn't have the time, so I am definitely behind this PR. :clap:

I agree, this is better overall to stop the possibility of type definition drifting in the future. I'm going to close #6 in favour of this.

File: package.json [hunk 1]
```diff
@@ -2,13 +2,19 @@
   "name": "graphql-subscriptions-client",
   "version": "0.13.0",
   "description": "A simpler client for graphql subscriptions based on apollographql/subscriptions-transport-ws",
-  "main": "client.js",
+  "main": "dist/index.js",
```
Comment (tom-fletcher) [inline package.json:5]: Could you add a module property pointing to the esm files? For me, the big advantage of this library is that it works with native ES module build tools, which needs to know about these.

```suggestion
  "module": "dist/esm//index.js",
  "main": "dist/index.js",
```

**Review by tom-fletcher — CHANGES_REQUESTED** [review]
See change in package.json

**Comment by OmgImAlexis** [comment]
I knew I had missed something. How's that?

**Comment by tom-fletcher** [comment]
@lunchboxer is it possible to have this merged and released?

**Review by lunchboxer — APPROVED** [review]
Looks good

**Pull request merged**